AzureUtils.sh

```diff
@@ -0,0 +1,220 @@
+    err_exit "Success!" NONE
+  fi
+
+  #  4. Waagent wants an ifcfg-eth0 config file
+  if [[ ! -f ${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0 ]]
+  then
+    err_exit "Creating ifcfg-eth0 file for network.service... " NONE
+    install -bDm 0644 -o root -g root /dev/null \
+      "${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0"
+    (
+      echo "DEVICE=eth0"
+      echo "ONBOOT=yes"
+      echo "BOOTPROTO=dhcp"
+      echo "TYPE=Ethernet"
+      echo "USERCTL=no"
+      echo "PEERDNS=yes"
+      echo "IPV6INIT=no"
+      echo "NM_CONTROLLED=no"
+    ) > "${CHROOTMNT}/etc/sysconfig/network-scripts/ifcfg-eth0" || \
+      err_exit "Failed creating ifcfg-eth0 file for network.service" 1
+    err_exit "Success!" NONE
+  fi
+
+  #  5. No static network-naming rules...
+  err_exit "Disabling static udev network-naming rules... " NONE
+  chroot "${AMIGENCHROOT}" ln -s /dev/null \
+    /etc/udev/rules.d/75-persistent-net-generator.rules ||
+    err_exit "Failed disabling static udev network-naming rules" 1
+  err_exit "Success!" NONE
+
+  #  6. Configure waagent for cloud-init
+  err_exit "Writing config-date to /etc/waagent.conf... " NONE
+  chroot "${AMIGENCHROOT}" sed -i \
+    -e 's/Provisioning.Agent=auto/Provisioning.Agent=auto/g' \
+    -e 's/ResourceDisk.Format=y/ResourceDisk.Format=n/g' \
+    -e 's/ResourceDisk.EnableSwap=y/ResourceDisk.EnableSwap=n/g' \
+    /etc/waagent.conf || \
+    err_exit "Failed writing config-date to /etc/waagent.conf" 1
+  err_exit "Success!" NONE
+
+  #  7. Allow only Azure datasource, disable fetching network setting via IMDS"
+  err_exit "Configure Azure datasource... " NONE
+  install -bDm 0644 -o root -g root <(
+    echo "datasource_list: [ Azure ]"
+    echo "datasource:"
+    echo "  Azure:"
+    echo "    apply_network_config: False"
+  ) "${AMIGENCHROOT}/etc/cloud/cloud.cfg.d/91-azure_datasource.cfg" || \
+    err_exit "Failed configuring Azure datasource" 1
+  err_exit "Success!" NONE
+
+  #  8. Add console log file
+  err_exit "Configuring console-logging for cloud-init... " NONE
+  install -bDm 0644 -o root -g root <(
+    echo "# This tells cloud-init to redirect its stdout and stderr to"
+    echo "# 'tee -a /var/log/cloud-init-output.log' so the user can see output"
+    echo "# there without needing to look on the console."
+    echo "output: {all: '| tee -a /var/log/cloud-init-output.log'}"
+  ) "${AMIGENCHROOT}/etc/cloud/cloud.cfg.d/05_logging.cfg" || \
+    err_exit "Failed configuring console-logging for cloud-init" 1
+  err_exit "Success!" NONE
+
+  #  9. Don't create swap space on the operating system disk.
+  err_exit "Configure swap-space creation..." NONE
+  chroot "${AMIGENCHROOT}" sed -i \
+    -e 's/ResourceDisk.Format=y/ResourceDisk.Format=n/g' \
+    -e 's/ResourceDisk.EnableSwap=y/ResourceDisk.EnableSwap=n/g' \
+    /etc/waagent.conf
+    err_exit "Failed configuring swap-space creation" 1
+  err_exit "Success!" NONE
+
```

Comment:
This is already present in Step 6.

Reply:
Down side of following the [vendor dox](https://learn.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-centos#centos-70): they're enumerated twice. Once in the "_Install cloud-init to handle the provisioning_" section and once in the "_Swap configuration_"  section. Sadly, no anchor tags to numbers 12 and 13 (respectively) in that linked-to section.